cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -275,20 +275,14 @@ function is_leader() {
   # Disabling because "${KUBECTL_OPTS}" needs to allow for expansion here
   KUBE_CONTROLLER_MANAGER_LEADER=$(${KUBECTL} ${KUBECTL_OPTS} -n kube-system get leases.v1.coordination.k8s.io kube-controller-manager -o "jsonpath={.spec.holderIdentity}")
 
-  case "${KUBE_CONTROLLER_MANAGER_LEADER}" in
-  "")
+  if [ -z "${KUBE_CONTROLLER_MANAGER_LEADER}" ]; then
     log ERR "No leader election info found."
     return 1
-    ;;
-
-  "${HOSTNAME}"_*)
+  elif [[ "${KUBE_CONTROLLER_MANAGER_LEADER}" =~ ^${HOSTNAME}_[a-zA-Z0-9-] || "${KUBE_CONTROLLER_MANAGER_LEADER}" =~ ^${HOSTNAME}---[a-zA-Z0-9-] ]]; then
     log INFO "Leader is $KUBE_CONTROLLER_MANAGER_LEADER"
     return 0
-    ;;
-
-  *)
-    log INFO "Leader is $KUBE_CONTROLLER_MANAGER_LEADER, not ${HOSTNAME}_*"
+  else
+    log INFO "Leader is $KUBE_CONTROLLER_MANAGER_LEADER, not ${HOSTNAME}_* or ${HOSTNAME}---*"
```

Comment:
```suggestion
    log INFO "Leader is ${KUBE_CONTROLLER_MANAGER_LEADER}, not ${HOSTNAME}_* or ${HOSTNAME}---<HASH>"
```

(missing braces are preexisting I know, but since we're here)

Reply:
Added